src/features/explicit-resource-management.md

```diff
@@ -0,0 +1,219 @@
+    }
+    const total = parseInt(contentLength, 10);
+    let loaded = 0;
+    
+    { // Introduce a new scope with a block
+        // Wrap the reader in a disposable resource
+        using readerResource = {
+        reader: response.body.getReader(),
+        [Symbol.dispose]() {
+            this.reader.releaseLock();
+        },
+        };
+        const { reader } = readerResource;
+    
+        let done = false;
+        let value;
+    
+        while (!done) {
+            ({ done, value } = await reader.read());
+            if (value) {
+            loaded += value.byteLength;
+            if (contentLength) {
+                const progress = (loaded / total) * 100;
+                console.log(`Downloaded ${loaded} of ${total} bytes (${progress.toFixed(2)}%)`);
+            }
+            updateProgressBar(progress); //some UI update
+            }
+        }
+    } // readerResource[Symbol.dispose]() is called automatically
+
+    // Do some other things... 
+    
+    { // Introduce a new scope with a block
+        // Get a new reader for the same stream
+        using readerResource = {
+        reader2: response.body.getReader(),
+        [Symbol.dispose]() {
+            this.reader2.releaseLock();
+        },
+        };
+        const { reader2 } = readerResource;
+    
+        let done2 = false;
+        let fileData = new Uint8Array();
+        while (!done2) {
+            const { done, value } = await reader2.read();
+            done2 = done;
+            if (value) {
+                const newFileData = new Uint8Array(fileData.length + value.length);
+                newFileData.set(fileData);
+                newFileData.set(value, fileData.length);
+                fileData = newFileData;
+                console.log(`Saving data to file... chunk length = ${value.length}`);
+            }
+        }
+
+    } // readerResource[Symbol.dispose]() is called automatically
+
+    ... // The rest of the code
+  }
+  
+  // Example usage:
+  downloadFileWithUsing('https://example.com/largefile.dat');
+```
+
+## `DisposableStack` and `AsyncDisposableStack`
+
+To further facilitate managing multiple disposable resources, the proposal introduces `DisposableStack` and `AsyncDisposableStack`. These stack-based structures allow developers to group and dispose of multiple resources in a coordinated manner. Resources are added to the stack, and when the stack is disposed, either synchronously or asynchronously, the resources are disposed of in the reverse order they were added, ensuring that any dependencies between them are handled correctly. This simplifies the cleanup process when dealing with complex scenarios involving multiple related resources. Both structures provide methods like `use()`, `adopt()`, and `defer()` to add resources or disposal actions, and a `dispose()` or `asyncDispose()` method to trigger the cleanup. They offer a robust way to manage the disposal of multiple resources within a defined scope.
```

Comment:
I think it should also be mentioned that these two kinds of objects have `[Symbol.dispose]` and `[Symbol.asyncDispose]`, respectively, so you can use them with `using`.